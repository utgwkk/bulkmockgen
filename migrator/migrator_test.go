package migrator

import (
	"strings"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestGenerate(t *testing.T) {
	testcases := []struct {
		name string
		m    *Migrator
	}{
		{
			name: "multi file",
			m: &Migrator{
				InputDir: "./fixtures/multifile",
			},
		},
	}
	for _, tc := range testcases {
		tc := tc
		t.Run(tc.name, func(t *testing.T) {
			var sb strings.Builder
			tc.m.writer = &sb

			err := tc.m.Migrate()
			if err != nil {
				t.Fatal(err)
			}

			snaps.MatchSnapshot(t, sb.String())
		})
	}
}
