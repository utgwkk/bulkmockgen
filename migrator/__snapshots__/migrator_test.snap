
[TestGenerate/multi_file - 1]
package multifile
// Code generated by mockgengen. DO NOT EDIT.

//go:generate go run github.com/utgwkk/mockgengen/cmd/mockgengen -package mock_bar -out ./mock_bar/mock_bar.go . MockBars
var MockBars = []any{
    new(IBar),
}
//go:generate go run github.com/utgwkk/mockgengen/cmd/mockgengen -package mock_foo -out ./mock_foo/mock_foo.go . MockFoos
var MockFoos = []any{
    new(IFoo),
}

---
